how-to/customize-workspace/client/src/framework/workspace/store.ts

```diff
@@ -407,3 +404,23 @@ async function getLandingPageRow(definition: StorefrontSettingsLandingPageRow, l
 		items: detailedNavigationItems
 	};
 }
+
+function addButtons(options: StorefrontProviderOptions, apps: PlatformApp[]): PlatformApp[] {
+	if (options.primaryButton || Array.isArray(options.secondaryButtons)) {
+		return apps.map((app) => ({
```

Comment:
Shouldn't this be the other way around?

If an app specifies a primary or secondary button then we don't override it with the default buttons.

If a common primary button is defined don't we need the same logic (i.e. pass the app to whatever action id is specified)?

Reply:
the first if just checks if there are any options set before processing, I have update to maintain existing primary or secondary if an app has them